openedx_atlas/sparse_checkout.py

```diff
@@ -0,0 +1,40 @@
+"""
+Performs sparse-checkout of subdirectories for a given <repository>:<branch>.
+
+Contains one function: sparse_checkout.
+"""
+import os
+import tempfile
+from shutil import copytree, ignore_patterns
+from subprocess import call
+
+
+def sparse_checkout(directory, repository, branch, output_path=os.getcwd()):
```

Comment:
What was the reason for doing this sparse checkout instead of copying the content directly using the [GitHub contents api](https://docs.github.com/en/rest/repos/contents#get-repository-content)?   Is there a goal to make this tool GitHub agnostic?